cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/LoadRunnable.java

```diff
@@ -64,13 +66,29 @@ protected BrokerStats getResult() throws Exception {
         return cachedBrokerStats;
       }
     }
+    if (_populateDiskInfo && !isClusterUsingJBOD()) {
+      throw new UserRequestException("populate_disk_info = true when Kafka cluster is NOT using JBOD");
+    }
+
     if (_start != DEFAULT_START_TIME_FOR_CLUSTER_MODEL) {
       return clusterModel(_modelCompletenessRequirements.minMonitoredPartitionsPercentage()).brokerStats(_kafkaCruiseControl.config());
     } else {
       return clusterModelFromEarliest().brokerStats(_kafkaCruiseControl.config());
     }
   }
 
+  private boolean isClusterUsingJBOD() throws Exception {
+    ClusterModel clusterModel = _kafkaCruiseControl.loadMonitor().clusterCapacity();
+    // If and only if all brokers in the cluster are using JBOD, the cluster is considered to be using JBOD
+    for (Broker broker : clusterModel.brokers()) {
+      if (!broker.isUsingJBOD()) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
Nit: This seems like a good candidate to replace with stream API for conciseness:
```
return clusterModel.brokers().stream().allMatch(Broker::isUsingJBOD);
```